news/models.py

```diff
@@ -18,27 +16,19 @@
 class ArticleManager(models.Manager):
 
     def published(self):
-        return self.filter(hidden=False).filter(
-            Q(pub_date=timezone.localtime().date(), pub_time__lt=timezone.localtime().time()) |
-            Q(pub_date__lt=timezone.localtime().date()))
+        return self.filter(hidden=False, publication_time__lte=timezone.localtime())
 
 
 class TimePlaceManager(models.Manager):
 
     def published(self):
-        return self.filter(hidden=False, event__hidden=False).filter(
-            Q(pub_date=timezone.localtime().date(), pub_time__lt=timezone.localtime().time()) |
-            Q(pub_date__lt=timezone.localtime().date()))
+        return self.filter(hidden=False, event__hidden=False).filter(publication_time__lte=timezone.localtime())
 
     def future(self):
-        return self.published().filter(
-            Q(end_date=timezone.localtime().date(), end_time__gt=timezone.localtime().time()) |
-            Q(end_date__gt=timezone.localtime().date()))
+        return self.published().filter(start_time__gte=timezone.localtime())
 
     def past(self):
-        return self.published().filter(
-            Q(end_date=timezone.localtime().date(), end_time__lt=timezone.localtime().time()) |
-            Q(end_date__lt=timezone.localtime().date()))
+        return self.published().filter(start_time__lt=timezone.localtime())
```

Comment:
Was it on purpose that `past()` and `future()` now filter relative to start instead of end time..? 😮

Reply:
No, it is just a mistake from on my part. End time is indeed better.